cmd/kube-apiserver/app/server.go

```diff
@@ -210,9 +212,16 @@ func Run(s *options.APIServer) error {
 				installSSH = instances.AddSSHKeyToAllInstances
 			}
 		}
-
+		if ports.KubeletReadOnlyPort <= 0 {
+			glog.Fatalf("Must enable kubelet read-only port if proxy ssh-tunneling is specified.")
+		}
 		// Set up the tunneler
-		tunneler = master.NewSSHTunneler(s.SSHUser, s.SSHKeyfile, installSSH)
+		healthCheckPath := &url.URL{
+			Scheme: "http",
+			Host:   net.JoinHostPort("127.0.0.1", strconv.Itoa(ports.KubeletReadOnlyPort)),
```

Comment:
the assumptions about this health check seem off to me... 127.0.0.1 doesn't seem right (does it work with kubelet in a container?), requiring the read-only port doesn't seem right (we're trying to remove unsecured access)